### PR TITLE
Better phrasing and code example for PDO connection management

### PR DESCRIPTION
**Pull request by colshrapnel**
The current phrasing appears to be incorrect. The Zend engine doesn't necessarily display the error and back trace. And catching exception is not required in order to hide the sensitive information.

### CODE REVIEW

File: reference/pdo/connections.xml [hunk 1]
```diff
@@ -36,28 +36,23 @@ $dbh = new PDO('mysql:host=localhost;dbname=test', $user, $pass);
...
-   explicitly (via a <literal>catch</literal> statement) or implicitly via
-   <function>set_exception_handler</function>.
+   If a <literal>PDOException</literal> object thrown from the PDO constructor
+   is not caught either explicitly (via a <literal>catch</literal> statement)
+   or implicitly via <function>set_exception_handler</function>, the default
+   action taken by the zend engine is to convert the exception into a fatal error
+   that contains a back trace. This back trace will likely reveal database
+   connection details. It is your responsibility to set the &php.ini; option
+   <link linkend="ini.display-errors"><parameter>display_errors</parameter></link>
+   to 0 on a live server as not to reveal this information.
```
Comment (Girgias) [inline reference/pdo/connections.xml:55]: I mean this is "just" standard behaviour. So maybe rephrase it to something like

```suggestion
   or implicitly via <function>set_exception_handler</function>.
   Otherwise, the default behaviour of converting an uncaught exception to a
   <constant>E_FATAL_ERROR</constant> will occur.
   The fatal error will contain a back trace that can leak connection details.
   As such, the &php.ini; option
   <link linkend="ini.display-errors"><parameter>display_errors</parameter></link>
   should be set to <literal>0</literal> on a production server.
```

**Comment by colshrapnel** [comment]
@Girgias thank you for the corrections. I tried my best to rephrase the rest.

File: reference/pdo/connections.xml [hunk 2]
```diff
@@ -36,28 +36,24 @@ $dbh = new PDO('mysql:host=localhost;dbname=test', $user, $pass);
...
-   explicitly (via a <literal>catch</literal> statement) or implicitly via
-   <function>set_exception_handler</function>.
+   Just like any other <link linkend="language.exceptions">exception</link>,
+   <literal>PDOException</literal> could be caught either explicitly, via
+   a &catch; statement, or implicitly via <function>set_exception_handler</function>.
+   Otherwise, the default behaviour of converting an uncaught exception to a
+   <constant>E_FATAL_ERROR</constant> will occur.
+   The fatal error will contain a back trace that can leak connection details.
+   As such, the &php.ini; option
+   <link linkend="ini.display-errors"><parameter>display_errors</parameter></link>
```
Comment (kamil-tekiela) [inline reference/pdo/connections.xml:55]:
```suggestion
   <link linkend="ini.display-errors"><literal>display_errors</literal></link>
```

File: reference/pdo/connections.xml [hunk 3]
```diff
@@ -36,28 +36,24 @@ $dbh = new PDO('mysql:host=localhost;dbname=test', $user, $pass);
...
-   the full database connection details, including the username and
-   password.  It is your responsibility to catch this exception, either
-   explicitly (via a <literal>catch</literal> statement) or implicitly via
-   <function>set_exception_handler</function>.
+   Just like any other <link linkend="language.exceptions">exception</link>,
+   <literal>PDOException</literal> could be caught either explicitly, via
+   a &catch; statement, or implicitly via <function>set_exception_handler</function>.
+   Otherwise, the default behaviour of converting an uncaught exception to a
+   <constant>E_FATAL_ERROR</constant> will occur.
+   The fatal error will contain a back trace that can leak connection details.
```
Comment (kamil-tekiela) [inline reference/pdo/connections.xml:53]:
```suggestion
   The fatal error will contain a backtrace that can leak connection details.
```
https://en.wikipedia.org/wiki/Stack_trace

File: reference/pdo/connections.xml [hunk 4]
```diff
@@ -36,28 +36,24 @@ $dbh = new PDO('mysql:host=localhost;dbname=test', $user, $pass);
...
   <para>
-   If your application does not catch the exception thrown from the PDO
-   constructor, the default action taken by the zend engine is to terminate
-   the script and display a back trace.  This back trace will likely reveal
-   the full database connection details, including the username and
-   password.  It is your responsibility to catch this exception, either
-   explicitly (via a <literal>catch</literal> statement) or implicitly via
-   <function>set_exception_handler</function>.
+   Just like any other <link linkend="language.exceptions">exception</link>,
+   <literal>PDOException</literal> could be caught either explicitly, via
```
Comment (kamil-tekiela) [inline reference/pdo/connections.xml:49]:
```suggestion
   <classname>PDOException</classname> could be caught either explicitly, via
```

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]
